packages/saved-views-react/src/applySavedView.ts

```diff
@@ -96,27 +208,26 @@ export async function applySavedView(
   settings: ApplySavedViewSettings | undefined = {},
   overrides?: DefaultExtensionHandlersApplyOverrides,
 ): Promise<void> {
-  if (settings.viewState !== "keep") {
-    // We use "hidden" as the default value for modelAndCategoryVisibilityFallback
-    // because users expect modelSelector.enabled and categorySelector.enabled to
-    // act as exclusive whitelists when modelSelector.disabled or categorySelector.disabled
-    // arrays are empty, respectively.
-    const { modelAndCategoryVisibilityFallback = "hidden" } = settings;
-    const viewState =
-      settings.viewState instanceof ViewState
-        ? settings.viewState
-        : await createViewState(iModel, savedViewData.viewData, {
-            modelAndCategoryVisibilityFallback,
-          });
-
-    if (settings.camera instanceof ViewPose) {
-      viewState.applyPose(settings.camera);
-    } else if (settings.camera === "keep") {
-      viewState.applyPose(viewport.view.savePose());
-    }
-
-    viewport.changeView(viewState, settings.viewChangeOptions);
-  }
+  let viewStateProps: ViewStateProps = await createSeedViewStateProps(
```

Comment:
@diegopinate , I could add an if to all of this to skip it if settings.viewState and settings.camera both equal "keep"--that way it would skip all of it when it's not needed. What do you think?

Reply:
I think I like the readability of it this way? Up to you

Reply:
Okay. I'll leave it as-is, then. I don't think it would be a very common use-case anyways.